Test/Unit/Helper/CreditmemoTest.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2023 Adyen NV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Test\Unit\Helper;
+
+use Adyen\Payment\Api\Data\OrderPaymentInterface;
+use Adyen\Payment\Helper\Data;
+use Adyen\Payment\Helper\Creditmemo;
+use Adyen\Payment\Model\CreditmemoFactory;
+use Adyen\Payment\Model\ResourceModel\Order\Payment;
+use Adyen\Payment\Test\Unit\AbstractAdyenTestCase;
+use Magento\Framework\App\Helper\Context;
+use Magento\Framework\Model\AbstractModel;
+
+class CreditmemoTest extends AbstractAdyenTestCase
+{
+    public function testCreateAdyenCreditMemo()
+    {
+        $pspReference = '12A34B56C78D910E';
+        $originalReference = 'E10D9C78B56A3412';
+        $refundAmount = 10.00;
+        $adyenOrderPaymentId = 5;
+
+        $paymentMock = $this->createConfiguredMock(\Magento\Sales\Model\Order\Payment::class, [
+            'getEntityId' => 5
+        ]);
+
+        $adyenOrderPaymentMock = $this->createConfiguredMock(OrderPaymentInterface::class, [
+            'getEntityId' => [OrderPaymentInterface::ENTITY_ID => $adyenOrderPaymentId]
+        ]);
+
+        $orderPaymentResourceModel = $this->createConfiguredMock(Payment::class, [
+            'getOrderPaymentDetails' => $adyenOrderPaymentMock->getEntityId()
+        ]);
+
+        $orderPaymentResourceModel->method('getOrderPaymentDetails')
+            ->with($originalReference, $paymentMock->getEntityId())
+            ->willReturn($adyenOrderPaymentMock->getEntityId());
+
+        $adyenCreditmemoMock = $this->createMock(\Adyen\Payment\Model\Creditmemo::class);
+        $adyenCreditmemoFactoryMock = $this->createGeneratedMock(CreditmemoFactory::class, ['create']);
+        $adyenCreditmemoFactoryMock->method('create')->willReturn($adyenCreditmemoMock);
+
+        $adyenCreditmemoResourceModel = $this->createMock(
+            \Adyen\Payment\Model\ResourceModel\Creditmemo\Creditmemo::class
```

Comment:
Some of the classes were used with full path and others were imported and used as short form. Could you please update and use import instead of full path?